### PR TITLE
getSubLayerProps: Use in all composite layers

### PR DESCRIPTION
**Pull request by georgios-uber**

### CODE REVIEW

File: src/core-layers/hexagon-layer/hexagon-layer.js [hunk 1]
```diff
@@ -347,13 +347,11 @@ export default class HexagonLayer extends CompositeLayer {
...
-    const {id, radius, elevationScale, extruded, coverage, lightSettings, fp64} = this.props;
-
-    const forwardProps = this.getBaseLayerProps();
+    const {radius, elevationScale, extruded, coverage, lightSettings, fp64} = this.props;
 
     // return props to the sublayer constructor
-    return Object.assign({}, forwardProps, {
-      id: `${id}-hexagon-cell`,
+    return super.getSubLayerProps({
+      id: 'hexagon-cell',
```
Comment (heshan0131) [inline src/core-layers/hexagon-layer/hexagon-layer.js]: I think id is necessary, when you have multiple Hexagon-layer, each hexagon-cell layer should have its unique id

Comment (georgios-uber) [reply src/core-layers/hexagon-layer/hexagon-layer.js]: getSubLayerProps() does this logic automatically.

Comment (ibgreen) [reply src/core-layers/hexagon-layer/hexagon-layer.js]: > I think id is necessary, when you have multiple Hexagon-layer, each hexagon-cell layer should have its unique id

This new method automatically prepends the parent layer id I think. That needs to be clear in the docs.

File: docs/api-reference/composite-layer.md [hunk 2]
```diff
@@ -78,9 +78,10 @@ functions called.
 The default implementation returns `pickParams.info` without any change.
 
 
-##### `getBaseLayerProps`
+##### `getSubLayerProps`
 
-Returns an object with the key base `Layer` props that `CompositeLayer`s typically forward to their sublayers.
```
Comment (ibgreen) [inline docs/api-reference/composite-layer.md]: We should mention the special handling of id and updateTriggers. This is new compared to the old function.

This function now also creates the complete sublayer props object, not just the core props.

File: src/core-layers/grid-layer/grid-layer.js [hunk 3]
```diff
@@ -330,7 +328,7 @@ export default class GridLayer extends CompositeLayer {
   }
 
   // for subclassing, override this method to return
-  // customized sub layer class
+  // customized sub layer classx
```
Comment (heshan0131) [inline src/core-layers/grid-layer/grid-layer.js]: typo?

File: src/core-layers/hexagon-layer/hexagon-layer.js (hunk 1 above)
Comment (heshan0131) [reply src/core-layers/hexagon-layer/hexagon-layer.js]: i see, make sense

**Pull request merged**